### PR TITLE
AP_ESC_Telem: Log RPM and raw RPM values as floats

### PR DESCRIPTION
**Pull request by mbuzdalov**
The values of RPM and raw RPM come to logging as `float`s, but were logged as `int32`s (multiplied by 100). This PR fixes it.

This PR is mutually exclusive with #25977.

[Log sample](https://github.com/ArduPilot/ardupilot/files/13932894/24-01-14_19-31-08_float_rpm.bin.tgz) is attached and looks OK.

### CODE REVIEW

**Review by IamPete1 — APPROVED** [review]
LGTM, Thanks!

File: libraries/AP_ESC_Telem/AP_ESC_Telem.cpp [hunk 1]
```diff
@@ -502,12 +502,12 @@ void AP_ESC_Telem::update()
 
                 float rpm = 0.0f;
```
Comment (IamPete1) [inline libraries/AP_ESC_Telem/AP_ESC_Telem.cpp:503]: At the risk of scope creeping you a second time now there floats we can use NaN. This means we can tell the difference between a genuine 0 rpm and unknown.

`float rpm = AP::logger().quiet_nanf()`

Comment (mbuzdalov) [reply libraries/AP_ESC_Telem/AP_ESC_Telem.cpp:503]: Around this place, there is also a problem of using `uint32_t` for microseconds, so that we erroneously mark one RPM as stale roughly every 35 minutes... so yet another tiny fix.

(I am currently working on Extended DShot Telemetry v2 which brings various status bits and stress levels from ESC, which will also introduce new stuff in this location, so I will be there for a while anyway...)

Comment (mbuzdalov) [reply libraries/AP_ESC_Telem/AP_ESC_Telem.cpp:503]: Just made a PR including this (#26252).
